force-app/main/default/triggers/ChangeOnQuoteSentStage.trigger

```diff
@@ -0,0 +1,23 @@
+trigger ChangeOnQuoteSentStage on Opportunity (After update) {
+    List<Task> taskList = new List<Task>(); 
+    for(Opportunity oppOld : Trigger.Old) {
```

Comment:
for loop in for loop - better to avoid this where possible.
you can update it to check and find oldStageName using following:
you can simple get map of the oldOpportunities within method Trigger.oldMap: Map<Id, Opportunity> oldMap = Trigger.oldMap;
and then take needed old record in the newrecords loop:
for(Opportunity oppNew : Trigger.New) {
   Opportunity oldRecord = oldMap.get(oppNew.Id);
}